text/0000-rust-lang-crates.md

```diff
@@ -0,0 +1,231 @@
+compatibility between users of the library externally and in `std`. In
+particular, traits and types defined in the crate are compatible across either
+way of importing them.
+
+### Deprecation
+
+At some point a library may become stale -- either because it failed to make it
+out of the nursery, or else because it was supplanted by a superior library. The
+libs subteam can deprecate nursery crates at any time, and can deprecate
+rust-lang crates through an RFC. This is expected to be a rare occurrence.
+
+Deprecated crates move to rust-lang-deprecated and are subsequently minimally
+maintained. Alternatively, if someone volunteers to maintain the crate,
+ownership can be transferred externally.
+
+## Advertising
+
+Part of the reason for having rust-lang crates is to have a clear, short list of
+libraries that are broadly useful, vetted and maintained. But where should this
+list appear?
+
+This RFC doesn't specify the complete details, but proposes a basic direction:
+
+- The crates in rust-lang should appear in the sidebar in the core rustdocs
+  distributed with Rust, along side the standard library. (For nightly releases,
+  we should include the nursery crates as well.)
+
+- The crates should also be published on crates.io, and should somehow be
+*badged*. But the design of a badging/curation system for crates.io is out of
+scope for this RFC.
+
+## Plan for existing crates
+
+There are already a number of non-`std` crates in rust-lang. Below, we give the
+full list along with recommended actions:
+
+### Transfer ownership
+
+Please volunteer if you're interested in taking one of these on!
+
+- rlibc
+- semver
+- threadpool
+
+### Move to rust-lang-nursery
+
+- bitflags
+- getopts
+- glob
+- libc
+- log
+- rand (note, @huonw has a major revamp in the works)
+- regex
+- rustc-serialize (but will likely be replaced by serde or other approach eventually)
+- tempdir (destined for `std` after reworking)
+- uuid
+
+### Move to rust-lang-deprecated
+
+- fourcc: highly niche
+- hexfloat: niche
+- num: this is essentially a dumping ground from 1.0 stabilization; needs a complete re-think.
+- term: API needs total overhaul
+- time: needs total overhaul destined for std
+- url: replaced by https://github.com/servo/rust-url
+
+# Drawbacks
+
+The drawbacks of this RFC are largely social:
+
+* Emphasizing rust-lang crates may alienate some in the Rust community, since it
+  means that certain libraries obtain a special "blessing". This is mitigated by
+  the fact that these libraries also become owned by the community at large.
+
+* On the other hand, requiring that ownership/governance be transferred to the
+  library subteam may be a disincentive for library authors, since they lose
+  unilateral control of their libraries. But this is an inherent aspect of the
+  policy design, and the vastly increased visibility of libraries is likely a
+  strong enough incentive to overcome this downside.
```

Comment:
Could we have joint ownership of promoted libs? It seems to really suck that you curate a library for some time, manage it, raise it like a child and then have to give up on it entirely. Could we keep the maintainer as an owner of the repo within rust-lang and have the lib managed by the maintainer and libs team together? Worst case scenario if consensus can't be reached the lib is booted to -deprecated (and either a new version proposed or it gets forked)